### ISSUE TITLE
If the dir of the path not exist, this plugin will not work.

### ISSUE THREAD

**Issue by vipcxj** [sysgears/webpack-virtual-modules#38]
For example:
```
/root
   node_module
   src
   test
      webpack.config.js
   package.json
```
run the webpack in test, and add a virtual path: `node_modules/a.json`. The webpack context is `test`, not `root`, so the webpack resolver will search `test/node_modules` first. however it does not exists. Then webpack think `a.json` is not in the directory `test/node_modules`, and will be unable to resolve it.

**Comment by vlasenko** [comment]
@vipcxj Please add link to repo with reproduction example

**Comment by vipcxj** [comment]
I have dig into the [resolver plugin code](https://github.com/webpack/enhanced-resolve/blob/master/lib/ModulesInHierachicDirectoriesPlugin.js#L33) provider by webpack.
In my example, the `webpack.resolve.modules` is `['node_modules']`, and `webpack.context` is '/path/to/project/test'. The request is `a.json`. Because the request is relative, the resolver will check the stats of directory `/path/to/project/test/node_modules`, and found it not exists. so it will check the stats of `/path/to/project/node_modules`. it exists. However the virtual path is `/path/to/project/test/node_modules/a.json`, so the webpack can not resolve the virtual module.

**Comment by vlasenko** [comment]
@vipcxj Please create public Git repo somewhere that has all the code needed to reproduce this issue

**Comment by vipcxj** [comment]
@vlasenko have created a pull request. with fix code and a test case. comment the fix code and run the test case to reproduce this issue

**Comment by vlasenko** [comment]
Closed via: #40
